src/sage/categories/rings.py

```diff
@@ -423,6 +423,28 @@ def is_integral_domain(self, proof=True) -> bool:
 
             return False
 
+        def is_integrally_closed(self) -> bool:
+            r"""
+            Return whether this ring is integrally closed.
+
+            This is the default implementation that returns ``False``.
+
+            EXAMPLES::
+
+                sage: x = polygen(ZZ, 'x')
+                sage: K.<a> = NumberField(x^2 + 189*x + 394)
+                sage: R = K.order(2*a)
+                sage: R.is_integrally_closed()
+                False
+                sage: R
+                Order of conductor 2 generated by 2*a in Number Field in a with defining polynomial x^2 + 189*x + 394
+                sage: S = K.maximal_order(); S
+                Maximal Order generated by a in Number Field in a with defining polynomial x^2 + 189*x + 394
+                sage: S.is_integrally_closed()
+                True
+            """
+            return False
```

Comment:
```suggestion
            raise NotImplementedError
```
And change the docstring accordingly.

Reply:
voila, voila